src/components/PopupWithForm.js

```diff
@@ -38,4 +38,6 @@ export default class PopupWithForm extends Popup {
       e.value = "";
     });
   }
+
+  // new code
```

Comment:
You should not close the popup in the submit handler. Close it in index.js in `then` block, so the modal won't be closed if an error occurred

Reply:
updated